portfolio/src/main/java/com/google/sps/servlets/HomeServlet.java

```diff
@@ -1,38 +1,86 @@
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.users.UserService;
 import com.google.appengine.api.users.UserServiceFactory;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.Optional;
 
 @WebServlet("/login")
 public class HomeServlet extends HttpServlet {
   private static final String HOME_PATH = "/";
+  private static final String USER_INFO = "userInfo";
+  private static final String ID = "id";
+  private static final String COMMENT_NICKNAME = "nickname";
+  // Hard-coded email of webpage admin.
+  private static final String ADMIN_EMAIL = "amytn@google.com";
+  private static final String NICKNAME_SERVLET = "/nickname";
+
+  private static String userEmail;
+  private static String nickname;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html");
 
     UserService userService = UserServiceFactory.getUserService();
-    if (userService.isUserLoggedIn()) {
-      String currentUserEmail = userService.getCurrentUser().getEmail();
-      String urlToRedirectToAfterUserLogsOut = HOME_PATH;
-      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
 
-      response.getWriter().println("<p>Hello " + currentUserEmail + "!</p>");
-      response.getWriter().println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
+
+    if (!userService.isUserLoggedIn()) {
+      String loginUrl = userService.createLoginURL(HOME_PATH);
+
+      response.getWriter().println("<style>#comment-form {display:none;}</style>");
+      response.getWriter().println("<p>Hello friend.</p>");
+      response.getWriter().println("<p>Please log in to Add a comment.</p>");
+      response.getWriter().println("<p>Login <a href=\"" + loginUrl + "\">here</a>.</p>");
       return;
     }
 
-    String urlToRedirectToAfterUserLogsIn = HOME_PATH;
-    String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
+    // If user has not set a nickname, redirect to nickname page
+    Optional<String> optionalNickname = getUserNickname(userService.getCurrentUser().getUserId());
```

Comment:
nit: "optionalNickname" can be renamed as "nickname". From the type you can tell it's optional.

Reply:
got it! 

done!